Proton/Sources/EditorCommand/TextProcessors/ListTextProcessor.swift

```diff
@@ -112,11 +112,18 @@ public class ListTextProcessor: TextProcessing {
 
         if let nextLine = editor.nextContentLine(from: currentLine.range.location),
             nextLine.range.endLocation < editor.contentLength - 1 {
-            let nextLineText = editor.attributedText.attributedSubstring(from: NSRange(location: nextLine.range.location, length: 1))
-            let paraStyle = nextLineText.attribute(.paragraphStyle, at: 0, effectiveRange: nil) as? NSParagraphStyle
-            let isFirstLevelListItem = (paraStyle?.firstLineHeadIndent ?? 0) / editor.listLineFormatting.indentation == 1
-            if nextLineText.attribute(.listItem, at: 0, effectiveRange: nil) != nil,
-                (nextLineText.string != "\n" || isFirstLevelListItem == false) {
+            let attributedText = editor.attributedText
+            var isNotNewLineCharacter: Bool {
```

Comment:
Pretty opinionated style 😉 I don't mind

Reply:
Yeah it's super opinionated. But what I like about it is that it doesn't pre-compute everything before the `if` statement. You can still make self-documenting variable names to make the `if` clauses are readable.